linalg/vector.hpp

```diff
@@ -143,28 +163,36 @@ class Vector
        in addition to the overloaded operator()(int). */
    inline operator const double *() const { return data; }
 
-   inline bool OwnsData() const { return (allocsize > 0); }
+   /// Return a reference to the Memory object used by the Vector.
+   Memory<double> &GetMemory() { return data; }
+
+   /** @brief Return a reference to the Memory object used by the Vector, const
+       version. */
+   const Memory<double> &GetMemory() const { return data; }
+
+   /// Read the Vector data (host pointer) ownership flag.
+   inline bool OwnsData() const { return data.OwnsHostPtr(); }
 
    /// Changes the ownership of the data; after the call the Vector is empty
    inline void StealData(double **p)
```

Comment:
Could we deprecate these Steal functions?

Reply:
Do you see an problem with having these?